File: FrontEnd/src/app/features/reservations/reservations/classes/dtos/form/reservation-write-dto.ts
import { Guid } from 'guid-typescript'
// Custom
import { PassengerWriteDto } from './passenger-write-dto'

export interface ReservationWriteDto {

    // PK
    reservationId: Guid
    // FKs
    customerId: number
    destinationId: number
    driverId?: number
    pickupPointId: number
    portId: number
    shipId?: number
    // Fields
    date: string
    refNo: string
    ticketNo: string
    email: string
    phones: string
    adults: number
    kids: number
    free: number
    remarks: string
    passengers: PassengerWriteDto[]
    // Rowversion
    putAt: string

}

File: API/Features/Reservations/Reservations/Dtos/ReservationWriteDto.cs
using System;
using System.Collections.Generic;
using API.Infrastructure.Interfaces;

namespace API.Features.Reservations.Reservations {

    public class ReservationWriteDto : IMetadata {

        // PK
        public Guid ReservationId { get; set; }
        // Fks
        public int CustomerId { get; set; }
        public int DestinationId { get; set; }
        public int PickupPointId { get; set; }
        public int PortId { get; set; }
        public int? DriverId { get; set; }
        public int? ShipId { get; set; }
        // Fields
        public string Date { get; set; }
        public DateTime Now { get; set; }
        public string RefNo { get; set; }
        public string TicketNo { get; set; }
        public string Email { get; set; }
        public string Phones { get; set; }
        public int Adults { get; set; }
        public int Kids { get; set; }
        public int Free { get; set; }
        public string Remarks { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }
        // Navigation
        public List<PassengerWriteDto> Passengers { get; set; }

    }

}